lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1810,13 +1810,18 @@ sub git_diff {
     return "\n" . $res->{stderr} if $res->{stderr};
 }
 
-=head2 investigate
+sub investigate {
+    my $self = shift;
+    return $self->{investigate} ||= $self->_investigate(@_);
```

Comment:
Why do you use this form of caching here? A new job object is instantiated for each db query so I don't thing this gives you any benefit.

Reply:
Having looked at the code, i'm not sure caching here does anything. You get a new job object for every request, and i don't see any code where you call `$job->investigate` in quick succession.

Reply:
when the user selects the tab, selects a different one and selects investigation again we call `$job->investigate` each time

Reply:
Ok, i don't see how that use case could benefit from this kind of caching (since each user interaction would result in a whole new HTTP request).

Reply:
You can just remove the caching, i'm sure now that it does nothing.

Reply:
That's what I mean with "A new job object is instantiated for each db query". If @kraih agrees we can assume that's correct.

> when the user selects the tab, selects a different one and selects investigation again we call $job->investigate each time

If you wanted to cover that case (and that's actually a good idea) you should simply cache the contents on the client-side. The contents of the tab are actually already preserved and the only improvement to made is that it doesn't make the AJAX query again. I can implement that quickly if you like.

Reply:
Yes, caching on client side is more important but also it does not cover everything. The more users select the investigation for a job the more we want to cache the hash in the server as well not to prevent DB queries but more so the filesystem accesses

Reply:
But if you implement in-memory caching do it right and don't keep cache entries around forever (until the next service restart). Reducing file system access reminds me that text results are currently stored within an extra file. This means a lot of file system access (resulting in long loading times) when opening test details of jobs with lots of such results (e.g. LTP tests). Maybe that's something we want to optimize first.

Reply:
Agreed, if we really benefit from server-side caching in this case (client-side caching is probably enough here) it's better to introduce a real caching module like `CHI`.